docs/ref/models/indexes.txt

```diff
@@ -49,6 +49,17 @@ The name of the index. If ``name`` isn't provided Django will auto-generate a
 name. For compatibility with different databases, index names cannot be longer
 than 30 characters and shouldn't start with a number (0-9) or underscore (_).
 
+.. admonition:: Partial indexes in abstract base classes
+
+    You must always specify a unique name for the index. This would normally
+    cause a problem for partial indexes in abstract base classes, since
+    ``name`` is required and the
+    :attr:`Meta.indexes <django.db.models.Options.indexes>` option on this
+    class are included into each of the child classes, with exactly the same
+    values for the attributes (including ``name``) each time. To work around
+    this problem, ``indexes`` option should be specified for each of the child
+    classes with unique names for indexes.
```

Comment:
Same as above:

```
    You must always specify a unique name for an index. As such, you
    cannot normally specify an index on an abstract base class, since the
    :attr:`Meta.indexes <django.db.models.Options.indexes>` option is
    inherited by subclasses, with exactly the same values for the attributes
    (including ``name``) each time. Instead, specify the ``indexes`` option
    on subclasses directly, providing a unique name for each index.
```